### PR TITLE
Week5

### PR DESCRIPTION
**Pull request by mishapark**
Sorry, forgot to merge starter project with the main branch. Please review code changes through my last commits

### CODE REVIEW

File: Week 5/Task App/Task App/Task_AppApp.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//
+
+import SwiftUI
+
+@main
+struct Task_AppApp: App {
+  var body: some Scene {
+    WindowGroup {
+      ContentView()
+        .environmentObject(Store())
```
Comment (rharshaw) [inline Week 5/Task App/Task App/Task_AppApp.swift:15]: Went the EnvironmentObject route, eh? Nice implementation.

File: Week 5/Task App/Task App/ContentView.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+//  Task App
+//
+//  Created by Mikhail Pak on 2023-09-27.
+//
+
+import SwiftUI
+
+struct ContentView: View {
+  @State var isAddTaskSheetOpen = false
+  @State var selectedTask: Task?
```
Comment (rharshaw) [inline Week 5/Task App/Task App/ContentView.swift:12]: I see this was something you thought of using. I'd like to know more. :)

File: Week 5/Task App/Task App/ContentView.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          }
+        }
+        VStack(alignment: .leading) {
+          Spacer()
+          NewTaskButton(onTap: {
+            isAddTaskSheetOpen = true
+          })
+        }
+      }
+      .padding(.horizontal)
```
Comment (rharshaw) [inline Week 5/Task App/Task App/ContentView.swift:32]: Nice UI work. You also can achieve the same result with a few less lines of code by changing the ZStack into a VStack and getting rid of the other VStack on line 25. Be sure to leave that Spacer() there and you got the same layout with less stacks. :)

File: Week 5/Task App/Task App/ViewModel/Store.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+import Foundation
+
+class Store: ObservableObject {
+  @Published var tasks: [Task] = [
+    Task(title: "Buy groceries", isCompleted: false, notes: "Bread, milk, eggs"),
+    Task(title: "Walk the dog", isCompleted: false, notes: "After work"),
+    Task(title: "Study Swift", isCompleted: true, notes: "Combine and RxSwift"),
+  ]
```
Comment (rharshaw) [inline Week 5/Task App/Task App/ViewModel/Store.swift:15]: Nice work! However, the homework asked for at least 5 to be listed upon the start. :)

File: Week 5/Task App/Task App/ViewModel/Store.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    if let newNote = note {
+      tasks[indexToUpdate].notes = newNote
+    }
+
+    if let newIsCompleted = isCompleted {
+      tasks[indexToUpdate].isCompleted = newIsCompleted
+    }
+  }
```
Comment (rharshaw) [inline Week 5/Task App/Task App/ViewModel/Store.swift:37]: Way to go above and beyond by creating an editing function! I see that you've done the Above and Beyond by disabling the 'Add' functionality if the title is empty. In this function, you can get around that by editing an already made task and eliminating the title. That'll allow the task to be saved with no title when they navigate back to the Home Screen. 

Consider adding some checks for the title if you want to get a little more in-depth and close that loophole. Up to you, though.  :)

File: Week 5/Task App/Task App/Views/NewTaskButton.swift [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+        Text("New Task")
+          .font(.headline)
+      })
+      .padding(.top)
+      Spacer()
+    }
+    .background(.white)
+  }
+}
```
Comment (rharshaw) [inline Week 5/Task App/Task App/Views/NewTaskButton.swift:27]: Nice button extraction!

File: Week 5/Task App/Task App/ContentView.swift (hunk 3 above)
Comment (mishapark) [reply Week 5/Task App/Task App/ContentView.swift:32]: Haven't thought about it, thanks!

File: Week 5/Task App/Task App/ContentView.swift (hunk 2 above)
Comment (mishapark) [reply Week 5/Task App/Task App/ContentView.swift:12]: As part of my last comment, that's the result of battling with StateObject lol

File: Week 5/Task App/Task App/Views/TaskRow.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        Spacer()
+        Image(systemName: task.isCompleted ? "checkmark.square" : "square")
+          .foregroundStyle(task.isCompleted ? .green : .red)
+          .fontWeight(.bold)
+          .font(.title3)
+          .onTapGesture {
+            withAnimation(.linear(duration: 0.2)) {
+              taskStore.updateTask(task, isCompleted: !task.isCompleted)
+            }
+          }
```
Comment (rharshaw) [inline Week 5/Task App/Task App/Views/TaskRow.swift:30]: Animations are always a nice touch.

**Review by rharshaw — APPROVED** [review]
This is really good! Nice use of EnvironmentObject; however, although it works, this is something that should be used to pass that data to views that you cannot directly pass down from a parent view. It's commonly used with stuff like passing in data from an authenticated user. 

Consider initializing the Store in your ContentView as a @StateObject and for all of your other views, you can use the @ObservedObject var taskStore: Store instead of the EnvironmentObject one. Then once that's done, just pass that data on down to the other views.

File: Week 5/Task App/Task App/Task_AppApp.swift (hunk 1 above)
Comment (rharshaw) [reply Week 5/Task App/Task App/Task_AppApp.swift:15]: Hahaha! I completely understand.

**Pull request merged**